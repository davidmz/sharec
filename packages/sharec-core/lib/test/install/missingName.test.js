const { fixtures } = require('testUtils')
const { vol } = require('memfs')
const { sharec } = require('../../')

describe('sharec > install > missing name', () => {
  const packageFxt = fixtures('package/json/08-install-missing-name')
  const babelFxt = fixtures('babel/json/00-base')
  const eslintFxt = fixtures('eslint/json/01-base')
  const npmignoreFxt = fixtures('npmignore/lines/00-base')
  const gitignoreFxt = fixtures('gitignore/lines/00-base')

  beforeEach(() => {
    jest.clearAllMocks()
    vol.reset()
  })

  it('installs configs even when target package name is missed', async () => {
    const semaphore = {
      start: jest.fn(),
      error: jest.fn(),
      success: jest.fn(),
      fail: jest.fn(),
    }
    const context = {
      targetPath: '/target',
      cache: {},
      configs: [],
      options: {
        cache: false,
      },
    }
    const dir = {
      '/target/.eslintrc': eslintFxt.current,
      '/target/.babelrc': babelFxt.current,
      '/target/.npmignore': npmignoreFxt.current,
      '/target/.gitignore': gitignoreFxt.current,
      '/target/package.json': packageFxt.current,
      '/target/node_modules/awesome-config/configs/.eslintrc': eslintFxt.upcoming,
      '/target/node_modules/awesome-config/configs/.editorconfig': 'bar',
      '/target/node_modules/awesome-config/configs/.babelrc': babelFxt.upcoming,
      '/target/node_modules/awesome-config/configs/.npmignore': npmignoreFxt.upcoming,
      '/target/node_modules/awesome-config/configs/.gitignore': gitignoreFxt.upcoming,
      '/target/node_modules/awesome-config/configs/package.json': packageFxt.upcoming,
      '/target/node_modules/awesome-config/package.json': JSON.stringify({
        name: 'awesome-config',
        version: '1.0.0',
      }),
    }
    vol.fromJSON(dir, '/')

    await sharec(context, semaphore)

    expect(vol.readFileSync('/target/.eslintrc', 'utf8')).toMatchSnapshot()
    expect(vol.readFileSync('/target/.babelrc', 'utf8')).toMatchSnapshot()
    expect(vol.readFileSync('/target/.editorconfig', 'utf8')).toMatchSnapshot()
    expect(vol.readFileSync('/target/.gitignore', 'utf8')).toMatchSnapshot()
    expect(vol.readFileSync('/target/.npmignore', 'utf8')).toMatchSnapshot()
    expect(vol.readFileSync('/target/package.json', 'utf8')).toMatchSnapshot()
  })
})
