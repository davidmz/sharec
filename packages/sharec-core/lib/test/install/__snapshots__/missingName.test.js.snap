// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sharec > install > missing name installs configs even when target package name is missed 1`] = `
"{
  \\"env\\": {
    \\"browser\\": true,
    \\"es6\\": true,
    \\"node\\": true,
    \\"jest\\": true
  },
  \\"globals\\": {
    \\"foo\\": true,
    \\"bar\\": true
  },
  \\"extends\\": \\"airbnb\\",
  \\"rules\\": {
    \\"max-len\\": [
      \\"warn\\",
      120,
      4,
      {
        \\"ignoreUrls\\": false,
        \\"ignoreComments\\": true,
        \\"ignoreStrings\\": false
      }
    ],
    \\"space-before-function-paren\\": 1,
    \\"no-console\\": 0
  },
  \\"parserOptions\\": {
    \\"parser\\": \\"babel-eslint\\",
    \\"ecmaVersion\\": 7,
    \\"ecmaFeatures\\": {
      \\"spread\\": true
    },
    \\"sourceType\\": \\"module\\"
  }
}
"
`;

exports[`sharec > install > missing name installs configs even when target package name is missed 2`] = `
"{
  \\"presets\\": [
    [
      \\"env\\",
      {
        \\"modules\\": false,
        \\"targets\\": {
          \\"browsers\\": [
            \\"> 1%\\",
            \\"last 2 versions\\",
            \\"not ie > 0\\"
          ]
        }
      }
    ],
    \\"stage-2\\"
  ],
  \\"plugins\\": [
    [
      \\"syntax-dynamic-import\\",
      {
        \\"foo\\": \\"bar\\"
      }
    ],
    \\"lodash\\"
  ],
  \\"env\\": {
    \\"test\\": {
      \\"presets\\": [
        \\"env\\",
        \\"stage-2\\"
      ],
      \\"plugins\\": [
        \\"dynamic-import-node\\"
      ]
    }
  },
  \\"ignore\\": [
    \\"*.babelrc\\"
  ]
}
"
`;

exports[`sharec > install > missing name installs configs even when target package name is missed 3`] = `
"bar
"
`;

exports[`sharec > install > missing name installs configs even when target package name is missed 4`] = `
"node_modules
foo
bar
baz
"
`;

exports[`sharec > install > missing name installs configs even when target package name is missed 5`] = `
"node_modules
foo
bar
baz
"
`;

exports[`sharec > install > missing name installs configs even when target package name is missed 6`] = `
"{
  \\"scripts\\": {
    \\"start\\": \\"rimraf && NODE_ENV=dev start-script\\",
    \\"commit\\": \\"git-cz\\",
    \\"lint\\": \\"eslint 'src/**/*.{js,jsx,json}'\\",
    \\"lint-staged\\": \\"lint-staged\\"
  },
  \\"dependencies\\": {
    \\"react\\": \\"16.0.0\\",
    \\"awesome-config\\": \\"1.0.0\\"
  },
  \\"sharec\\": {
    \\"configs\\": [
      \\"awesome-config\\"
    ],
    \\"locked\\": {
      \\"awesome-config\\": \\"1.0.0\\"
    }
  },
  \\"lint-staged\\": {
    \\"src/**/*.{js,jsx,json}\\": [
      \\"eslint --fix\\",
      \\"prettier --write\\",
      \\"git add\\"
    ]
  },
  \\"husky\\": {
    \\"hooks\\": {
      \\"pre-commit\\": \\"lint-staged\\",
      \\"commit-msg\\": \\"commitlint -E HUSKY_GIT_PARAMS\\"
    }
  },
  \\"config\\": {
    \\"commitizen\\": {
      \\"path\\": \\"cz-conventional-changelog\\"
    }
  },
  \\"prettier\\": {
    \\"semi\\": false,
    \\"singleQuote\\": true,
    \\"trailingComma\\": \\"all\\"
  },
  \\"eslintConfig\\": {
    \\"parser\\": \\"babel-eslint\\",
    \\"extends\\": [
      \\"standard\\",
      \\"prettier\\",
      \\"prettier/standard\\"
    ],
    \\"plugins\\": [
      \\"prettier\\"
    ],
    \\"env\\": {
      \\"jest\\": true,
      \\"node\\": true
    },
    \\"rules\\": {
      \\"prettier/prettier\\": [
        \\"error\\",
        {
          \\"semi\\": false,
          \\"singleQuote\\": true,
          \\"trailingComma\\": \\"all\\"
        }
      ]
    }
  },
  \\"eslintIgnore\\": [
    \\"/node_modules\\"
  ],
  \\"devDependencies\\": {
    \\"@commitlint/cli\\": \\"^7.6.1\\",
    \\"@commitlint/config-conventional\\": \\"^7.6.0\\",
    \\"babel-eslint\\": \\"^10.0.1\\",
    \\"commitizen\\": \\"^3.1.1\\",
    \\"cz-conventional-changelog\\": \\"^2.1.0\\",
    \\"eslint\\": \\"^5.16.0\\",
    \\"eslint-config-prettier\\": \\"^4.2.0\\",
    \\"eslint-config-prettier-standard\\": \\"^2.0.0\\",
    \\"eslint-config-standard\\": \\"^12.0.0\\",
    \\"eslint-plugin-import\\": \\"^2.17.2\\",
    \\"eslint-plugin-node\\": \\"^9.0.1\\",
    \\"eslint-plugin-prettier\\": \\"^3.1.0\\",
    \\"eslint-plugin-promise\\": \\"^4.1.1\\",
    \\"eslint-plugin-standard\\": \\"^4.0.0\\",
    \\"husky\\": \\"^2.2.0\\",
    \\"lint-staged\\": \\"^8.1.6\\",
    \\"prettier\\": \\"^1.17.1\\"
  }
}"
`;
